datafusion/common/src/column.rs

```diff
@@ -71,7 +71,11 @@ impl Column {
         }
     }
 
-    fn from_idents(idents: &mut Vec<String>) -> Option<Self> {
+    /// Create a Column from multiple normalized identifiers
+    ///
+    /// For example, `foo.bar` would be represented as a two element vector
+    /// `["foo", "bar"]`
+    pub fn from_idents(mut idents: Vec<String>) -> Option<Self> {
```

Comment:
i made this public and cleaned up the signature so I could reuse it when dealing with the changes to `USING` planning

However I am not  sure what a `USING(foo.bar)` would actually mean 🤔  Maybe when joining across multiple schemas? 

I would be interested in hearing the reviewers opinions on if I should should revert this change and return a planner error if it got a multi-part ObjectName in a `USING` clause 

Perhaps @jonahgao  has some opinions here

Reply:
This syntax is from Snowflake. I tested it on Snowflake, and `USING(foo.bar)` has the same effect as  `USING(bar)`.
![CleanShot 2025-01-24 at 10 57 25@2x](https://github.com/user-attachments/assets/95da4ee6-8f6c-49c9-960c-78ca246947b8)

DataFusion gave a different result; it did not deduplicate the using column.
```sh
> with t1 as (select 1 as a),
t2 as (select 1 as a)
select * from t1 join t2 using(t1.a);
+---+---+
| a | a |
+---+---+
| 1 | 1 |
+---+---+.
```
Instead of giving an incorrect result, perhaps the simplest approach is to return an error.

Reply:
Thank you @jonahgao  -- I will make this change

Reply:
I have made such queries error now and added a test in [250115a](https://github.com/apache/datafusion/pull/14255/commits/250115ac5c1805cf3324ba94bb613bdb5171403e)